sylt-compiler/src/typechecker.rs

```diff
@@ -1210,25 +1233,36 @@ impl TypeChecker {
                 }
 
                 (Type::Blob(a_blob, a_fields), Type::Blob(b_blob, b_fields)) => {
-                    let mut c_fields = BTreeMap::new();
-                    for (a_name, a_ty) in a_fields.iter() {
-                        let b_ty = match b_fields.get(a_name) {
+                    if a_fields.len() != b_fields.len() {
+                        return err_type_error!(
+                            self,
+                            span,
+                            ctx,
+                            TypeError::Mismatch {
+                                got: self.bake_type(a),
+                                expected: self.bake_type(b)
+                            },
+                            "Blobs have different number of fields"
+                        );
+                    }
```

Comment:
I dislike this error message. We can do better by pointing to the actual fields that are wrong. This will not really help anyone. :(
In a perfect world we'd search through the fields and look for a close word, and then add that as a guess. It will save me not noticing typos. :3

Copy the bellow for-loop and reverse it. It has the added benefit of making the code symetric.

Reply:
We do that in the expression. Perhaps it is a good idea to do it here too. Ideally we would only show the differences. I don't think we thought about that when we first implemented it. There might be a nice solution.